src/middlewared/middlewared/plugins/chart_releases_linux/chart_release.py

```diff
@@ -560,15 +560,20 @@ async def do_update(self, job, chart_release, data):
 
         await self.perform_actions(context)
 
+        # When a user edits a chart release, pods will be scaled up again.
+        # This will cause pods to spin up even for a few seconds, before we have a chance to scale it down,
+        # if the previous status was STOPPED. By adding in the context the "desiredStatus" as "STOPPED",
+        # we are letting the app developer to take extra action if wanted. Like explicitly setting the replica count to 0.
         config = await add_context_to_configuration(config, {
             CONTEXT_KEY_NAME: {
                 **get_action_context(chart_release),
                 'operation': 'UPDATE',
+                'desiredStatus': release_orig['status'],
                 'isUpdate': True,
             }
         }, self.middleware, chart_release)
 
-        await self.middleware.call('chart.release.helm_action', chart_release, chart_path, config, 'update')
+        await self.middleware.call('chart.release.helm_action', chart_release, chart_path, config, 'upgrade')
```

Comment:
This does not change functionality, just a typo, it was falling into the else case anyway

https://github.com/truenas/middleware/blob/e1f3091ced8c94ab9035a62ff20f8240ec8109c4/src/middlewared/middlewared/plugins/chart_releases_linux/helm.py#L27

Reply:
Please keep it as it was before because it is used in error message and in truenas context it is update and not upgrade which is bound to cause confusion